File: src/main/java/com/examportal/dto/ExamResponseDTO.java
package com.examportal.dto;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ExamResponseDTO extends BaseDTO {
    private String title;
    private String subject;
    private Integer totalMarks;
    private Integer durationMin;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private List<QuestionResponseDTO> questions = new ArrayList<>();
}

File: Dotnet3/ExamPortal/DTOs/ExamResponseDTO.cs
namespace ExamPortal.DTOs
{
    public class ExamResponseDTO
    {
        public long Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Subject { get; set; } = string.Empty;
        public int TotalMarks { get; set; }
        public int DurationMin { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public int QuestionCount { get; set; }
    }
}
